metadata/src/main/java/org/apache/kafka/metadata/migration/MigrationManifest.java

```diff
@@ -60,7 +61,8 @@ public MigrationManifest build() {
             if (endTimeNanos == 0) {
                 endTimeNanos = time.nanoseconds();
             }
-            return new MigrationManifest(total, batches, endTimeNanos - startTimeNanos, counts);
+            Map<MetadataRecordType, Integer> orderedCounts = new TreeMap<>(counts);
```

Comment:
qq: why we need a treemap here? Is the sorted order help here?

Reply:
Putting into a treemap will order the map according to the keys. The effect of this is that the log message becomes deterministic, which is useful for testing.